[docs] Improve `DataGrid` migration guide (#7861)
mui/mui-x: docs/data/migration/migration-data-grid-v5/migration-data-grid-v5.md

## Run codemods

The `preset-safe` codemod will automatically adjust the bulk of your code to account for breaking changes in v6. You can run `v6.0.0/data-grid/preset-safe` targeting only Data Grid or `v6.0.0/preset-safe` to target Date and Time pickers as well.

> It should be only applied **once per folder.**

Author: I think this doesn't make sense as almost all our codemods (at least preset-safe ones) are supposed to be `idempotent` so this line may confuse the user in case of some errors on the first run. It should be OK to run it any number of time as user wishes.

CC @LukasTy @alexfauquette

Reviewer: Good point. 👍 
It indeed makes little sense having this sentence, especially including **bold**. 🙈

npx @mui/x-codemod v6.0.0/preset-safe <path>
```

:::success

> Not taking into consideration the removals of the methods and exports, as they should be simple and require manual intervention, around 50% of the DataGrid breaking changes are automatically handled by the `preset-safe` codemod. 🎉

Reviewer 2: ```suggestion
Apart from the removed methods and exports that require manual intervention, around 50% of the DataGrid breaking changes are automatically handled by the `preset-safe` codemod 🎉.
```

To make this long sentence a bit shorter.

npx @mui/x-codemod v6.0.0/preset-safe <path>
```

> If we don't take into consideration the removals of the methods and exports, which should be simple and require manual intervention, around 50% of the breaking changes are automatically handled by the `preset-safe` codemod. 🎉

Author: Or a better place for sharing such stats would be blog post?

    }}
  />
  ```

- The `componentError` event was removed. Use the [error boundary](https://reactjs.org/docs/error-boundaries.html) to catch errors thrown during rendering.
- The `GridCallbackDetails['api']` was removed from event details. Use the `apiRef` returned by `useGridApiContext` or `useGridApiRef` instead.
- The `cellFocusIn` and `cellFocusOut` events are internal now. Use `componentsProps.cell.onFocus` and `componentsProps.cell.onBlur` props instead.

:::info
To know more about the supported events and their signatures, check the [events catalog](https://next.mui.com/x/react-data-grid/events/#catalog-of-events) page.
:::

### Columns

> - The `GridColDef['hide']` property was removed. Use `GridColDef['columnVisibility']` instead.

Author: I couldn't find `GridColDef['columnVisibility']`, probably it's removed?

Reviewer 2: It has never existed 👌
